.github/workflows/integration.yml

```diff
@@ -0,0 +1,44 @@
+name: integration
+
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+
+jobs:
+  install-production:
+    runs-on: ${{ matrix.os }}
+    permissions:
+      contents: read
+    
+    strategy:
+      matrix:
+        node-version: [14, 16, 18]
+        os: [ubuntu-latest]
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      
+      - name: Install Production
+        run: |
+          npm install --legacy-deps --legacy-peer-deps --production
```

Comment:
I'm not extremely familiar with the exact behavior of the first two flags you're providing, but if we're trying to reproduce the scenario of a user doing `npm i fastify`, shall we not just specify only `--production`?

Reply:
I think the same about using bash and I don't want to "pollute" the project. 

 Since we are using  `--production`  and `tap` is a `devDependency`, it will not be available in a straightforward way.

About `--legacy-deps` and `legacy-peer-deps`, it was suggested on the issue.

Reply:
let's see what @jsumners thinks about the flags as he suggested them in the issue. wrt tap, you can just run tests using npx tap or, I believe you can simply run tap tests using node (but I'm not exactly sure if tap needs to be installed, in fact I realize now I'm not really sure how that works!)

Reply:
Yes, we need the flags in order to prevent npm from installing a flat dependency tree. The primary thing we are trying to catch in this initial server based check is transitive dependencies vs direct dependencies: Fastify should be listing all of its direct dependencies and not relying on transitive dependencies.

We should go with the Bash route to avoid installing _any_ development dependencies. This will avoid tainting the environment (another reason to have the independent workflow file).

Given that this is mainly a smoke test for dependencies, we can limit the run environment to `ubuntu-latest` + `lts/*`.

Reply:
Also, yes, `tap` would have to be installed to even do `node some-tap-based.js`.

Reply:
@jsumners ok I see your point about using tap as it may install coincidentally a dependency that's missing but if we use npx it will be installed in a temporary location which will not affect the working of the fastify app we're testing, so I wonder if it's really worth the trouble using bash and curl instead

About the additional flags, I still don't understand. Shall we not reproduce the same scenario as somebody installing fastify? Why would providing those flags would make any difference?

Reply:
What is the trouble with Bash and curl?

`--legacy-deps` prevents npm from hoisting transitive dependencies to the top level of the `node_modules` directory.

`--legacy-peer-deps` prevents automatic install of peer dependencies throughout the tree.

Again, the purpose of this task is to verify that Fastify's code is not referencing transitive dependencies. These flags put npm into its most strict dependency resolution mode, thereby covering all bases. We could also `pnpm install` and get the same effect.